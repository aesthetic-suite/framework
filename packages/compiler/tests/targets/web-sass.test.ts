import runTargetTests from './runTargetTests';

runTargetTests('web', 'web-sass');
