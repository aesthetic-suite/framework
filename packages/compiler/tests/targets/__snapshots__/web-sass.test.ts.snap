// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format web-sass compiles and writes files 1`] = `
"// Auto-generated by Aesthetic. Do not modify!

@mixin border-sm
  border-color: $palette-neutral-color-30
  border-radius: $border-sm-radius
  border-style: solid
  border-width: $border-sm-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin border-df
  border-color: $palette-neutral-color-30
  border-radius: $border-df-radius
  border-style: solid
  border-width: $border-df-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin border-lg
  border-color: $palette-neutral-color-30
  border-radius: $border-lg-radius
  border-style: solid
  border-width: $border-lg-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin box-sm
  background-color: $palette-neutral-color-10
  border-color: $palette-neutral-color-30
  border-radius: $border-sm-radius
  border-style: solid
  border-width: $border-sm-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin box-df
  background-color: $palette-neutral-color-10
  border-color: $palette-neutral-color-30
  border-radius: $border-df-radius
  border-style: solid
  border-width: $border-df-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin box-lg
  background-color: $palette-neutral-color-10
  border-color: $palette-neutral-color-30
  border-radius: $border-lg-radius
  border-style: solid
  border-width: $border-lg-width

  &:focus
    border-color: $palette-primary-bg-focused

  &:hover
    border-color: $palette-primary-bg-hovered

  &:active
    border-color: $palette-primary-bg-selected

  &[disabled]
    border-color: $palette-primary-bg-disabled


@mixin heading-l1
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l1-letter-spacing
  line-height: $heading-l1-line-height
  font-size: $heading-l1-size


@mixin heading-l2
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l2-letter-spacing
  line-height: $heading-l2-line-height
  font-size: $heading-l2-size


@mixin heading-l3
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l3-letter-spacing
  line-height: $heading-l3-line-height
  font-size: $heading-l3-size


@mixin heading-l4
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l4-letter-spacing
  line-height: $heading-l4-line-height
  font-size: $heading-l4-size


@mixin heading-l5
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l5-letter-spacing
  line-height: $heading-l5-line-height
  font-size: $heading-l5-size


@mixin heading-l6
  font-family: $typography-font-heading
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  letter-spacing: $heading-l6-letter-spacing
  line-height: $heading-l6-line-height
  font-size: $heading-l6-size


@mixin pattern-hidden
  display: none


@mixin pattern-offscreen
  clip-path: rect(1px, 1px, 1px, 1px)
  height: 1
  left: -5vw
  overflow: hidden
  position: fixed
  width: 1


@mixin pattern-reset-button
  appearance: none
  background-color: transparent
  border: 0
  cursor: pointer
  display: inline-block
  font-size: inherit
  margin: 0
  padding: 0
  text-decoration: none
  user-select: auto
  vertical-align: middle


@mixin pattern-reset-input
  appearance: none
  background-color: transparent
  margin: 0
  padding: 0
  width: 100%

  &::-moz-focus-outer
    border: 0


@mixin pattern-reset-list
  list-style: none
  margin: 0
  padding: 0


@mixin pattern-reset-typography
  font-family: inherit
  font-weight: normal
  word-wrap: break-word


@mixin pattern-text-break
  overflow-wrap: break-word
  word-wrap: break-word
  word-break: break-word


@mixin pattern-text-truncate
  overflow: hidden
  text-overflow: ellipsis
  white-space: nowrap


@mixin pattern-text-wrap
  overflow-wrap: normal
  word-wrap: normal
  word-break: normal


@mixin shadow-xs
  box-shadow: $shadow-xs-x $shadow-xs-y $shadow-xs-blur $shadow-xs-spread $palette-neutral-color-90


@mixin shadow-sm
  box-shadow: $shadow-sm-x $shadow-sm-y $shadow-sm-blur $shadow-sm-spread $palette-neutral-color-90


@mixin shadow-md
  box-shadow: $shadow-md-x $shadow-md-y $shadow-md-blur $shadow-md-spread $palette-neutral-color-90


@mixin shadow-lg
  box-shadow: $shadow-lg-x $shadow-lg-y $shadow-lg-blur $shadow-lg-spread $palette-neutral-color-90


@mixin shadow-xl
  box-shadow: $shadow-xs-x $shadow-xs-y $shadow-xs-blur $shadow-xs-spread $palette-neutral-color-90


@mixin text-sm
  font-family: $typography-font-text
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  font-size: $text-sm-size
  line-height: $text-sm-line-height


@mixin text-df
  font-family: $typography-font-text
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  font-size: $text-df-size
  line-height: $text-df-line-height


@mixin text-lg
  font-family: $typography-font-text
  font-weight: normal
  word-wrap: break-word
  color: $palette-neutral-color-80
  font-size: $text-lg-size
  line-height: $text-lg-line-height


"
`;

exports[`Format web-sass compiles and writes files 2`] = `
"// Auto-generated by Aesthetic. Do not modify!

$border-sm-radius: 0.11rem !default // 1.50px
$border-sm-width: 0rem !default // 0px
$border-df-radius: 0.21rem !default // 3px
$border-df-width: 0.07rem !default // 1px
$border-lg-radius: 0.32rem !default // 4.50px
$border-lg-width: 0.14rem !default // 2px

$breakpoint-xs-query: '(min-width: 45.71em)' !default
$breakpoint-xs-query-size: 640 !default
$breakpoint-xs-root-line-height: 1.33 !default
$breakpoint-xs-root-text-size: 14.94px !default
$breakpoint-sm-query: '(min-width: 68.57em)' !default
$breakpoint-sm-query-size: 960 !default
$breakpoint-sm-root-line-height: 1.42 !default
$breakpoint-sm-root-text-size: 15.94px !default
$breakpoint-md-query: '(min-width: 91.43em)' !default
$breakpoint-md-query-size: 1280 !default
$breakpoint-md-root-line-height: 1.52 !default
$breakpoint-md-root-text-size: 17.01px !default
$breakpoint-lg-query: '(min-width: 114.29em)' !default
$breakpoint-lg-query-size: 1600 !default
$breakpoint-lg-root-line-height: 1.62 !default
$breakpoint-lg-root-text-size: 18.15px !default
$breakpoint-xl-query: '(min-width: 137.14em)' !default
$breakpoint-xl-query-size: 1920 !default
$breakpoint-xl-root-line-height: 1.73 !default
$breakpoint-xl-root-text-size: 19.36px !default

$depth-content: 100 !default
$depth-navigation: 1000 !default
$depth-sheet: 1100 !default
$depth-overlay: 1200 !default
$depth-modal: 1300 !default
$depth-toast: 1400 !default
$depth-dialog: 1500 !default
$depth-menu: 1600 !default
$depth-tooltip: 1700 !default

$heading-l1-letter-spacing: 0.25px !default
$heading-l1-line-height: 1.5 !default
$heading-l1-size: 1.14rem !default // 16px
$heading-l2-letter-spacing: 0.33px !default
$heading-l2-line-height: 1.69 !default
$heading-l2-size: 1.43rem !default // 20px
$heading-l3-letter-spacing: 0.44px !default
$heading-l3-line-height: 1.9 !default
$heading-l3-size: 1.79rem !default // 25px
$heading-l4-letter-spacing: 0.59px !default
$heading-l4-line-height: 2.14 !default
$heading-l4-size: 2.23rem !default // 31.25px
$heading-l5-letter-spacing: 0.79px !default
$heading-l5-line-height: 2.4 !default
$heading-l5-size: 2.79rem !default // 39px
$heading-l6-letter-spacing: 1.05px !default
$heading-l6-line-height: 2.7 !default
$heading-l6-size: 3.49rem !default // 48.80px

$shadow-xs-x: 0rem !default // 0px
$shadow-xs-y: 0.14rem !default // 2px
$shadow-xs-blur: 0.14rem !default // 2px
$shadow-xs-spread: 0rem !default // 0px
$shadow-sm-x: 0rem !default // 0px
$shadow-sm-y: 0.23rem !default // 3.24px
$shadow-sm-blur: 0.27rem !default // 3.75px
$shadow-sm-spread: 0rem !default // 0px
$shadow-md-x: 0rem !default // 0px
$shadow-md-y: 0.37rem !default // 5.24px
$shadow-md-blur: 0.39rem !default // 5.50px
$shadow-md-spread: 0rem !default // 0px
$shadow-lg-x: 0rem !default // 0px
$shadow-lg-y: 0.61rem !default // 8.47px
$shadow-lg-blur: 0.52rem !default // 7.25px
$shadow-lg-spread: 0rem !default // 0px
$shadow-xl-x: 0rem !default // 0px
$shadow-xl-y: 0.98rem !default // 13.71px
$shadow-xl-blur: 0.64rem !default // 9px
$shadow-xl-spread: 0rem !default // 0px

$spacing-xs: 0.31rem !default // 4.38px
$spacing-sm: 0.63rem !default // 8.75px
$spacing-df: 1.25rem !default // 17.50px
$spacing-md: 2.50rem !default // 35px
$spacing-lg: 3.75rem !default // 52.50px
$spacing-xl: 5rem !default // 70px
$spacing-unit: 17.5 !default

$text-sm-line-height: 1.25 !default
$text-sm-size: 0.89rem !default // 12.45px
$text-df-line-height: 1.25 !default
$text-df-size: 1rem !default // 14px
$text-lg-line-height: 1.25 !default
$text-lg-size: 1.13rem !default // 15.75px

$typography-font-heading: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"' !default
$typography-font-monospace: '\\"Lucida Console\\", Monaco, monospace' !default
$typography-font-text: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"' !default
$typography-font-system: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"' !default
$typography-root-line-height: 1.25 !default
$typography-root-text-size: 14px !default
"
`;

exports[`Format web-sass compiles and writes files 3`] = `
"// Auto-generated by Aesthetic. Do not modify!
@import '../example'

// contrast: normal
// scheme: light

$palette-brand-color-00: #000 !default
$palette-brand-color-10: #000 !default
$palette-brand-color-20: #000 !default
$palette-brand-color-30: #000 !default
$palette-brand-color-40: #000 !default
$palette-brand-color-50: #000 !default
$palette-brand-color-60: #000 !default
$palette-brand-color-70: #000 !default
$palette-brand-color-80: #000 !default
$palette-brand-color-90: #000 !default
$palette-brand-bg-base: #000 !default
$palette-brand-bg-disabled: #000 !default
$palette-brand-bg-focused: #000 !default
$palette-brand-bg-hovered: #000 !default
$palette-brand-bg-selected: #000 !default
$palette-brand-fg-base: #000 !default
$palette-brand-fg-disabled: #000 !default
$palette-brand-fg-focused: #000 !default
$palette-brand-fg-hovered: #000 !default
$palette-brand-fg-selected: #000 !default

$palette-primary-color-00: #000 !default
$palette-primary-color-10: #000 !default
$palette-primary-color-20: #000 !default
$palette-primary-color-30: #000 !default
$palette-primary-color-40: #000 !default
$palette-primary-color-50: #000 !default
$palette-primary-color-60: #000 !default
$palette-primary-color-70: #000 !default
$palette-primary-color-80: #000 !default
$palette-primary-color-90: #000 !default
$palette-primary-bg-base: #000 !default
$palette-primary-bg-disabled: #000 !default
$palette-primary-bg-focused: #000 !default
$palette-primary-bg-hovered: #000 !default
$palette-primary-bg-selected: #000 !default
$palette-primary-fg-base: #000 !default
$palette-primary-fg-disabled: #000 !default
$palette-primary-fg-focused: #000 !default
$palette-primary-fg-hovered: #000 !default
$palette-primary-fg-selected: #000 !default

$palette-secondary-color-00: #000 !default
$palette-secondary-color-10: #000 !default
$palette-secondary-color-20: #000 !default
$palette-secondary-color-30: #000 !default
$palette-secondary-color-40: #000 !default
$palette-secondary-color-50: #000 !default
$palette-secondary-color-60: #000 !default
$palette-secondary-color-70: #000 !default
$palette-secondary-color-80: #000 !default
$palette-secondary-color-90: #000 !default
$palette-secondary-bg-base: #000 !default
$palette-secondary-bg-disabled: #000 !default
$palette-secondary-bg-focused: #000 !default
$palette-secondary-bg-hovered: #000 !default
$palette-secondary-bg-selected: #000 !default
$palette-secondary-fg-base: #000 !default
$palette-secondary-fg-disabled: #000 !default
$palette-secondary-fg-focused: #000 !default
$palette-secondary-fg-hovered: #000 !default
$palette-secondary-fg-selected: #000 !default

$palette-tertiary-color-00: #000 !default
$palette-tertiary-color-10: #000 !default
$palette-tertiary-color-20: #000 !default
$palette-tertiary-color-30: #000 !default
$palette-tertiary-color-40: #000 !default
$palette-tertiary-color-50: #000 !default
$palette-tertiary-color-60: #000 !default
$palette-tertiary-color-70: #000 !default
$palette-tertiary-color-80: #000 !default
$palette-tertiary-color-90: #000 !default
$palette-tertiary-bg-base: #000 !default
$palette-tertiary-bg-disabled: #000 !default
$palette-tertiary-bg-focused: #000 !default
$palette-tertiary-bg-hovered: #000 !default
$palette-tertiary-bg-selected: #000 !default
$palette-tertiary-fg-base: #000 !default
$palette-tertiary-fg-disabled: #000 !default
$palette-tertiary-fg-focused: #000 !default
$palette-tertiary-fg-hovered: #000 !default
$palette-tertiary-fg-selected: #000 !default

$palette-neutral-color-00: #000 !default
$palette-neutral-color-10: #000 !default
$palette-neutral-color-20: #000 !default
$palette-neutral-color-30: #000 !default
$palette-neutral-color-40: #000 !default
$palette-neutral-color-50: #000 !default
$palette-neutral-color-60: #000 !default
$palette-neutral-color-70: #000 !default
$palette-neutral-color-80: #000 !default
$palette-neutral-color-90: #000 !default
$palette-neutral-bg-base: #000 !default
$palette-neutral-bg-disabled: #000 !default
$palette-neutral-bg-focused: #000 !default
$palette-neutral-bg-hovered: #000 !default
$palette-neutral-bg-selected: #000 !default
$palette-neutral-fg-base: #000 !default
$palette-neutral-fg-disabled: #000 !default
$palette-neutral-fg-focused: #000 !default
$palette-neutral-fg-hovered: #000 !default
$palette-neutral-fg-selected: #000 !default

$palette-muted-color-00: #000 !default
$palette-muted-color-10: #000 !default
$palette-muted-color-20: #000 !default
$palette-muted-color-30: #000 !default
$palette-muted-color-40: #000 !default
$palette-muted-color-50: #000 !default
$palette-muted-color-60: #000 !default
$palette-muted-color-70: #000 !default
$palette-muted-color-80: #000 !default
$palette-muted-color-90: #000 !default
$palette-muted-bg-base: #000 !default
$palette-muted-bg-disabled: #000 !default
$palette-muted-bg-focused: #000 !default
$palette-muted-bg-hovered: #000 !default
$palette-muted-bg-selected: #000 !default
$palette-muted-fg-base: #000 !default
$palette-muted-fg-disabled: #000 !default
$palette-muted-fg-focused: #000 !default
$palette-muted-fg-hovered: #000 !default
$palette-muted-fg-selected: #000 !default

$palette-danger-color-00: #000 !default
$palette-danger-color-10: #000 !default
$palette-danger-color-20: #000 !default
$palette-danger-color-30: #000 !default
$palette-danger-color-40: #000 !default
$palette-danger-color-50: #000 !default
$palette-danger-color-60: #000 !default
$palette-danger-color-70: #000 !default
$palette-danger-color-80: #000 !default
$palette-danger-color-90: #000 !default
$palette-danger-bg-base: #000 !default
$palette-danger-bg-disabled: #000 !default
$palette-danger-bg-focused: #000 !default
$palette-danger-bg-hovered: #000 !default
$palette-danger-bg-selected: #000 !default
$palette-danger-fg-base: #000 !default
$palette-danger-fg-disabled: #000 !default
$palette-danger-fg-focused: #000 !default
$palette-danger-fg-hovered: #000 !default
$palette-danger-fg-selected: #000 !default

$palette-warning-color-00: #000 !default
$palette-warning-color-10: #000 !default
$palette-warning-color-20: #000 !default
$palette-warning-color-30: #000 !default
$palette-warning-color-40: #000 !default
$palette-warning-color-50: #000 !default
$palette-warning-color-60: #000 !default
$palette-warning-color-70: #000 !default
$palette-warning-color-80: #000 !default
$palette-warning-color-90: #000 !default
$palette-warning-bg-base: #000 !default
$palette-warning-bg-disabled: #000 !default
$palette-warning-bg-focused: #000 !default
$palette-warning-bg-hovered: #000 !default
$palette-warning-bg-selected: #000 !default
$palette-warning-fg-base: #000 !default
$palette-warning-fg-disabled: #000 !default
$palette-warning-fg-focused: #000 !default
$palette-warning-fg-hovered: #000 !default
$palette-warning-fg-selected: #000 !default

$palette-success-color-00: #000 !default
$palette-success-color-10: #000 !default
$palette-success-color-20: #000 !default
$palette-success-color-30: #000 !default
$palette-success-color-40: #000 !default
$palette-success-color-50: #000 !default
$palette-success-color-60: #000 !default
$palette-success-color-70: #000 !default
$palette-success-color-80: #000 !default
$palette-success-color-90: #000 !default
$palette-success-bg-base: #000 !default
$palette-success-bg-disabled: #000 !default
$palette-success-bg-focused: #000 !default
$palette-success-bg-hovered: #000 !default
$palette-success-bg-selected: #000 !default
$palette-success-fg-base: #000 !default
$palette-success-fg-disabled: #000 !default
$palette-success-fg-focused: #000 !default
$palette-success-fg-hovered: #000 !default
$palette-success-fg-selected: #000 !default

$palette-info-color-00: #000 !default
$palette-info-color-10: #000 !default
$palette-info-color-20: #000 !default
$palette-info-color-30: #000 !default
$palette-info-color-40: #000 !default
$palette-info-color-50: #000 !default
$palette-info-color-60: #000 !default
$palette-info-color-70: #000 !default
$palette-info-color-80: #000 !default
$palette-info-color-90: #000 !default
$palette-info-bg-base: #000 !default
$palette-info-bg-disabled: #000 !default
$palette-info-bg-focused: #000 !default
$palette-info-bg-hovered: #000 !default
$palette-info-bg-selected: #000 !default
$palette-info-fg-base: #000 !default
$palette-info-fg-disabled: #000 !default
$palette-info-fg-focused: #000 !default
$palette-info-fg-hovered: #000 !default
$palette-info-fg-selected: #000 !default
"
`;
